build_library/catalyst.sh

```diff
@@ -207,58 +198,50 @@ catalyst_init() {
     # so far so good, expand path to work with weird comparison code below
     FLAGS_seed_tarball=$(readlink -f "$FLAGS_seed_tarball")
 
-    if [[ ! "$FLAGS_seed_tarball" =~ .*\.tar\.bz2 ]]; then
-        die_notrace "Seed tarball doesn't end in .tar.bz2 :-/"
+    if [[ ! "$FLAGS_seed_tarball" =~ .\.tar\.(bz2|xz) ]]; then
+        die_notrace "Seed tarball doesn't end in .tar.bz2 or .tar.xz :-/"
     fi
 
     # catalyst is obnoxious and wants the $TYPE/stage3-$VERSION part of the
     # path, not the real path to the seed tarball. (Because it could be a
     # directory under $TEMPDIR instead, aka the SEEDCACHE feature.)
     if [[ "$FLAGS_seed_tarball" =~ "$CATALYST_ROOT/builds/".* ]]; then
         SEED="${FLAGS_seed_tarball#$CATALYST_ROOT/builds/}"
-        SEED="${SEED%.tar.bz2}"
+        SEED="${SEED%.tar.*}"
     else
         mkdir -p "$CATALYST_ROOT/builds/seed"
         cp -n "$FLAGS_seed_tarball" "$CATALYST_ROOT/builds/seed"
         SEED="seed/${FLAGS_seed_tarball##*/}"
-        SEED="${SEED%.tar.bz2}"
+        SEED="${SEED%.tar.*}"
     fi
 }
 
 write_configs() {
     info "Creating output directories..."
-    mkdir -m 775 -p "$TEMPDIR/portage/repos.conf" "$DISTDIR"
+    mkdir -m 775 -p "$DISTDIR"
     chown portage:portage "$DISTDIR"
     info "Writing out catalyst configs..."
     info "    catalyst.conf"
     catalyst_conf > "$TEMPDIR/catalyst.conf"
     info "    catalystrc"
     catalystrc > "$TEMPDIR/catalystrc"
-    info "    portage/repos.conf/coreos.conf"
-    repos_conf > "$TEMPDIR/portage/repos.conf/coreos.conf"
     info "    stage1.spec"
     catalyst_stage1 > "$TEMPDIR/stage1.spec"
-    info "    stage2.spec"
-    catalyst_stage2 > "$TEMPDIR/stage2.spec"
-    info "    stage3.spec"
-    catalyst_stage3 > "$TEMPDIR/stage3.spec"
-    info "    stage4.spec"
-    catalyst_stage4 > "$TEMPDIR/stage4.spec"
```

Comment:
why did you move the spec generation from `write_configs()` to `catalyst_build()`?

Reply:
You used to be able to set `source_subpath` via a command line option, but you can't anymore. The `SEED` variable changes depending on which stages were built earlier, so now it's easiest to write the spec when it's actually needed.